src/main/java/xyz/holocons/mc/holoitemsrevamp/enchantment/Momento.java

```diff
@@ -0,0 +1,79 @@
+package xyz.holocons.mc.holoitemsrevamp.enchantment;
+
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Player;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.format.TextDecoration;
+import xyz.holocons.mc.holoitemsrevamp.HoloItemsRevamp;
+import xyz.holocons.mc.holoitemsrevamp.ability.BlockPlace;
+import xyz.holocons.mc.holoitemsrevamp.ability.PlayerDeath;
+import xyz.holocons.mc.holoitemsrevamp.enchant.CustomEnchantment;
+
+public class Momento extends CustomEnchantment implements PlayerDeath, BlockPlace {
+
+    private final HoloItemsRevamp plugin;
+
+    public Momento(HoloItemsRevamp plugin) {
+        super(plugin, "momento");
+        this.plugin = plugin;
+    }
+
+    @Override
+    public int getMaxLevel() {
+        return 1;
+    }
+
+    @Override
+    public boolean conflictsWith(@NotNull Enchantment other) {
+        return false;
+    }
+
+    @Override
+    public boolean canEnchantItem(@NotNull ItemStack item) {
+        return item.getType() == Material.ENDER_CHEST;
+        // Is there a better way to do this?
+        // Not sure why, but I feel like this can be done better.
+    }
+
+    @Override
+    public @NotNull Component displayName(int level) {
+        return Component.text()
+            .color(NamedTextColor.DARK_PURPLE)
+            .decoration(TextDecoration.ITALIC, false)
+            .append(Component.text("Momento"))
+            .build();
+    }
+
+    @Override
+    public int getItemStackCost(ItemStack itemStack) {
+        return Integer.MAX_VALUE;
+    }
+
+    @Override
+    public void run(PlayerDeathEvent event, ItemStack itemStack) {
+        // Set keepInv and keepExp on (For this event and player only)
```

Comment:
If `PlayerDeathEvent::getKeepInventory is true already, return.

Also use `PlayerDeathEvent::setShouldDropExperience` instead of `EntityDeathEvent::setDroppedExp`.

Reply:
Would a similar check be needed for `getKeepLevel`?

Reply:
Just checked, nope.